plugin/cog/cog-streams/src/main/java/it/geosolutions/imageioimpl/plugins/cog/CachingCogImageInputStream.java

```diff
@@ -153,7 +153,7 @@ public CogTileInfo getCogTileInfo() {
      * There are likely lots of optimizations to be made in here.
      */
     @Override
-    public void readRanges() {
+    public void readRanges(CogTileInfo cogTileInfo) {
```

Comment:
The class has a CogTileInfo field, is it still needed now? And the getter for it? Same for DefaultCogImageInputStream.java

Reply:
That CogTileInfo stores the header. fields and getter updated